libraries/dotnet/EnergyOriginEventStore/EnergyOriginEventStore/EventStore/FlatFile/FlatFileEventConsumer.cs

```diff
@@ -0,0 +1,110 @@
+using EnergyOriginEventStore.EventStore.Internal;
+using EnergyOriginEventStore.EventStore.Serialization;
+
+namespace EnergyOriginEventStore.EventStore.FlatFile;
+
+public class FlatFileEventConsumer : IEventConsumer
+{
+    private readonly FlatFileEventStore _fileStore;
+    private readonly IUnpacker _unpacker;
+    private readonly long _fromIssued;
+    private readonly long _fromFraction;
+
+    private readonly FileSystemWatcher _rootWatcher;
+    private readonly List<FileSystemWatcher> _watchers;
+    private readonly Dictionary<Type, IEnumerable<Action<Event<EventModel>>>> _handlers;
+
+    public FlatFileEventConsumer(IUnpacker unpacker, Dictionary<Type, IEnumerable<Action<Event<EventModel>>>> handlers, FlatFileEventStore fileStore, string topicPrefix, string? pointer)
+    {
+        _fileStore = fileStore;
+        _unpacker = unpacker;
+        _handlers = handlers;
+
+        if (pointer is not null)
+        {
+            try
+            {
+                var a = pointer.Split('-');
+                _fromIssued = long.Parse(a[0]);
+                _fromFraction = long.Parse(a[1]);
+            }
+            catch (Exception)
+            {
+                throw new InvalidDataException($"Pointer '{pointer}' not a valid format");
+            }
+        }
+
+        Console.WriteLine($"* Setup watchers using: {fileStore.ROOT} and prefix: {topicPrefix}");
```

Comment:
Change to use ILogger<>

Reply:
This would require us to handle dependency injection in this library. This can be added at a later point